### PR TITLE
Add rule to check the number of tasks

### PR DESCRIPTION
**Pull request by Ruchip16**
Introduce a new experimental rule that checks that number of tasks per file is less than
100.

Related: #2173
Fixes: #2172

### CODE REVIEW

**Review by ssbarnea — CHANGES_REQUESTED** [review]
@Ruchip16 Please be sure that you address all the linting problems, `tox -e lint` should pass before the effective review starts.

**Comment by Ruchip16** [comment]
> @Ruchip16 Please be sure that you address all the linting problems, `tox -e lint` should pass before the effective review starts.

sure, i will keep in mind

**Review by ssbarnea — CHANGES_REQUESTED** [review]
- [ ] Add a complexity.md file with documentation about rule
- [ ] Create a symlink from docs/rules/complexity.md -> the one in source code
- [ ] Add tests inside complexity.py file like in other rules (pytest)
- [x] Update description field
- [x] Update version_added and make it "v6.15.0"
- [x] Update tags to include "experimental" and "idiom"

There should be two test files named:
- `examples/playbooks/rule-complexity-pass.yml`
- `examples/playbooks/rule-complexity-fail.yml`

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
I tried to get this in and uncovered a longer chain of issues, so I am moving it draft until I do the preparatory work to get it in.

**Comment by ssbarnea** [comment]
Yes, please.

File: .vscode/settings.json
Comment (ssbarnea) [inline .vscode/settings.json:1]: Do not touch this file as part of this change.

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**